### PR TITLE
feat: Ignore invalidated atoms in useAtomsSnapshot

### PR DESCRIPTION
**Pull request by Aslemammad**
Task: invalidating atoms in #911 
Tries to ignore the snapshots of invalidated atoms, as inspired by useAtomsDevtools!

### CODE REVIEW

**Review by dai-shi — COMMENTED** [review]
We can merge this into the branch, but we will see how it goes after #1255.

File: src/devtools/useAtomsSnapshot.ts
Comment (Aslemammad) [reply src/devtools/useAtomsSnapshot.ts:31]: That's for sure!

**Review by dai-shi — APPROVED** [review]
Merging this and will have a second review in #1311.

**Pull request merged**
